cmake/target_toolchain.cmake

```diff
@@ -30,7 +30,7 @@ set(CMAKE_SYSTEM_VERSION ${PROJECT_VERSION})
 # We are not building dynamically loadable libraries
 set(BUILD_SHARED_LIBS OFF)
 
-if(NOT (COMPILER STREQUAL "host-gcc"))
+if(NOT ((COMPILER STREQUAL "host-gcc") OR (COMPILER STREQUAL "host-clang")))
```

Comment:
how is host-clang different from the current clang compiler definition we have?

Reply:
@nashif 

> how is host-clang different from the current clang compiler definition we have?

Not too much. After the discussion with @SebastianBoe, I am in progress of merging this with the existing cmake/toolchain/llvm and cmake/compiler/clang.

However I am missing something subtle there and can’t get it to work that way yet.

The original need is to use libFuzzer to fuzz a few states in the experimental TCP compiled as a native_posix application.